### PR TITLE
Ce/app status 2

### PR DESCRIPTION
**Pull request by calellowitz**
@snopoke @benrudolph 
Cal fixes app status report, take 2
This depends on https://github.com/dimagi/commcare-hq/pull/16219 and must be deployed after

### CODE REVIEW

File: corehq/apps/users/models.py [hunk 1]
```diff
@@ -796,9 +796,24 @@ class LastSubmission(DocumentSchema):
...
+    build_version = IntegerProperty()
+    app_id = StringProperty()
+
 
+class LastBuild(DocumentSchema):
+    build_version = IntegerProperty()
+    build_version_date = DateTimeProperty()
+    app_id = StringProperty()
+
+class ReportingMetadata(DocumentSchema):
```
Comment (dimagimon) [inline corehq/apps/users/models.py]: E302 expected 2 blank lines, found 1

File: corehq/apps/reports/standard/deployments.py [hunk 2]
```diff
@@ -59,113 +59,145 @@ class ApplicationStatusReport(DeploymentsReport):
...
     @property
     @memoized
     def selected_app_id(self):
         return self.request_params.get(SelectApplicationFilter.slug, None)
 
-    @property
-    @memoized
-    def users(self):
-        mobile_user_and_group_slugs = self.request.GET.getlist(LocationRestrictedMobileWorkerFilter.slug)
+    @quickcache(['app_id'], timeout=10 * 60)
```
Comment (benrudolph) [inline corehq/apps/reports/standard/deployments.py]: can cache this for way longer i suspect. i don't think it matters if the name is wrong

File: corehq/apps/reports/standard/deployments.py [hunk 3]
```diff
@@ -59,113 +59,145 @@ class ApplicationStatusReport(DeploymentsReport):
...
+                    last_sync = self.get_data_for_app(last_syncs, self.selected_app_id)
+                    if last_sync is None:
+                        last_sync = self.get_data_for_app(last_syncs, None)
+                last_builds = user.get('reporting_metadata', {}).get('last_builds')
+                if last_builds:
+                    last_build = self.get_data_for_app(last_builds, self.selected_app_id)
+            else:
+                last_sub = user.get('reporting_metadata', {}).get('last_submission_for_user', {})
+                last_sync = user.get('reporting_metadata', {}).get('last_sync_for_user', {})
+                last_build = user.get('reporting_metadata', {}).get('last_build', {})
```
Comment (benrudolph) [inline corehq/apps/reports/standard/deployments.py]: can dry this up a bit by getting `reporting_metatdata` to a separate var

File: corehq/apps/users/models.py [hunk 4]
```diff
@@ -796,9 +796,24 @@ class LastSubmission(DocumentSchema):
...
     last_submissions = SchemaListProperty(LastSubmission)
+    last_submission_for_user = SchemaProperty(LastSubmission)
+    last_syncs = SchemaListProperty(LastSync)
+    last_sync_for_user = SchemaProperty(LastSync)
+    last_builds = SchemaListProperty(LastBuild)
+    last_build = SchemaProperty(LastBuild)
```
Comment (benrudolph) [inline corehq/apps/users/models.py]: ouf

**Review by benrudolph — APPROVED** [review]

**Review by snopoke — APPROVED** [review]

File: corehq/apps/reports/standard/deployments.py [hunk 5]
```diff
@@ -59,115 +58,159 @@ class ApplicationStatusReport(DeploymentsReport):
...
+                             alt_prop_name='reporting_metadata.last_sync_for_user.sync_date'),
             DataTablesColumn(_("Application"),
-                             help_text=_("Displays application of last submitted form")),
+                             help_text=_("Displays application of last submitted form"),
+                             prop_name='reporting_metadata.last_builds.app_id',
+                             alt_prop_name='reporting_metadata.build_version_app'),
             DataTablesColumn(_("Application Version"),
-                             help_text=_("Displays application version of the user's last sync."),
-                             sort_type=DTSortType.NUMERIC),
+                             help_text = _("Displays application version of the user's last sync."),
```
Comment (dimagimon) [inline corehq/apps/reports/standard/deployments.py]: E251 unexpected spaces around keyword / parameter equals

File: corehq/apps/reports/standard/deployments.py [hunk 6]
```diff
@@ -59,115 +58,157 @@ class ApplicationStatusReport(DeploymentsReport):
...
+                             prop_name='reporting_metadata.last_syncs.sync_date',
+                             alt_prop_name='reporting_metadata.last_sync_for_user.sync_date'),
             DataTablesColumn(_("Application"),
-                             help_text=_("Displays application of last submitted form")),
+                             help_text=_("Displays application of last submitted form"),
+                             sortable=False),
             DataTablesColumn(_("Application Version"),
-                             help_text=_("Displays application version of the user's last sync."),
-                             sort_type=DTSortType.NUMERIC),
+                             help_text = _("Displays application version of the user's last sync."),
```
Comment (dimagimon) [inline corehq/apps/reports/standard/deployments.py]: E251 unexpected spaces around keyword / parameter equals

**Comment by calellowitz** [comment]
@snopoke @benrudolph all reindexes done, this is ready to go out.

**Comment by snopoke** [comment]
:+1: once build passes

**Comment by calellowitz** [comment]
✅

**Comment by benrudolph** [comment]
💥

**Pull request merged**